### DISCUSSION TITLE
`ProjectReference` has a incorrect version in nuget package

### DISCUSSION THREAD

**Discussion by SongOfYouth** [GitTools/GitVersion#3378] [Q&A] [answered]
**Describe the bug**
when a project A references another project B and both use the `GitVersion`,  the nuget package of A always has a version `1.0.0.0` dependency for B.

## Expected Behavior
the dependency version is same with B.

**Comment by HHobeck** [comment]
Could you please give us more information about what you are trieng to achieve? Generally GitVersion is a tool to generate a semantic version based on the state of a git repository. Your question seems to me out of the scope of GitVersion. It would be very helpful to find the root course of your issue before creating a bug.

**Reply by SongOfYouth** [reply]
the problem is, when ProjA referenced ProjB by `ProjectReference `, the nuget package created by ProjA has a dependency with the nuget which created by ProjB by default. But the tagert version is always `1.0.0`, if the real version of B is '0.2.x.x', threre is a error 'can't find the package  B with version '1.0.0'' when references package A in other project by `PackageReference`.

**Reply by SongOfYouth** [reply]
it may be a problem of nuget, but I can't find the way to fix it.

**Reply by HHobeck** [reply, marked as answer]
Normally you are referencing the nuget package directly instead of a project reference if it is an external library. But if you referencing the project itself it means it is part of your repository and part of your versioning strategy which will be determined (on a build server) using GitVersion. In that case the version will be given from GitVersion to the compiler so the referenced project has the desired version reference as well.

Happy Branching!
